src/srvpool.c

```diff
@@ -335,7 +335,7 @@ natsSrvPool_addNewURLs(natsSrvPool *pool, const natsUrl *curUrl, char **urls, in
             s = _addURLToPool(pool, url, true, tlsName);
         }
         natsStrHashIter_Done(&iter);
-        if ((s == NATS_OK) && added && pool->randomize)
+        if ((s == NATS_OK) && *added && pool->randomize)
```

Comment:
Good catch, but that means that our tests were not catching that. I believe we have pool randomize tests, but I guess not one that ensured that the pool was not randomized when getting an URL we already had.